### ISSUE TITLE
API documentation

### ISSUE THREAD

**Issue by chriscamacho** [lexbor/lexbor#159]
There doesn't at least seem to be any doxygen tags, or manually created API documentation.

A library this size is really to big to be looking at examples and guestimating how its supposed to be used.

Adding even brief doxygen comments to the source would go a long way to meeting the end user half way,,,

**Comment by vtorri** [comment]
that is something I would also like to have

**Comment by lexborisov** [comment]
@chriscamacho 

Yes, you are absolutely right, it is necessary to create documentation. Otherwise, there is not much point in all this code.
I'll think about it.

**Comment by vtorri** [comment]
@lexborisov I know you like to rewrite everything, but doxygen is a very good tool to add doc in header files. It is widely available and used. I also know it quite well. And it is not a strong dependency to the library. You can also tweak the output so that it embeds quite well in an existing web site.

**Comment by vtorri** [comment]
Doxygen can:
 * create documentation as html or pdf
 * create pages to document the design, etc.. (could be in separate files too)
 * add example in comments
 * integrate nicely the doc of the api in a website
 *  and other nice stuff that i can't remember

**Comment by chriscamacho** [comment]
having used doxygen in a previous project I can also recommend it, in the mean time I ended up using libml2 which I rather suspect is harder to use than this library - but I can't tell for sure... as I've not been able to evaluate it.

**Comment by GeneGajewski** [comment]
Very nice. awesome HTML library! But, yeah, documentation is holding it back.
